auth_totp/wizards/res_users_authenticator_create.py

```diff
@@ -73,8 +73,10 @@ def _compute_qr_code_tag(self):
                 continue
 
             totp = pyotp.TOTP(record.secret_key)
+            user_dname = urllib.quote(
```

Comment:
This URL encoding step is not needed because PyOTP already [calls ``quote``](https://github.com/pyotp/pyotp/blob/master/src/pyotp/utils.py#L61) as part of the URI creation process (this is why UTF-8 is not supported in the first place). Calling it twice causes the name to show up incorrectly in my MFA app because it is not fully decoded.

Reply:
Ahh good ol' Unicode.

Wouldn't we just want to pass `record.user_id.display_name.encode('utf-8')` to `totp.provisioning_url`?

Reply:
That said, this encoding would be specific to the underlying encoding defined for the PSQL database, right?

Reply:
@lasley - I don't think the DB encoding matters since the relevant Odoo field should always return a ``unicode`` object or raise an exception, and a ``unicode`` object can always be encoded using UTF-8 (in v9, the logic for ensuring the former is [here](https://github.com/odoo/odoo/blob/9.0/openerp/loglevels.py)).

A bigger question is whether the user's MFA device/app will be able to properly decode UTF-8. I tried this with LastPass Authenticator and Duo Mobile, which both worked fine, and I suspect that most will. However, if we're really worried about compatibility, we could encode to ASCII or Latin1 and only fall back to UTF-8 if that fails.